source/customization/menus.rst

```diff
@@ -1,5 +1,367 @@
+
+
+.. _menu-merging:
+
+Menu Merging
+============
+
+``.sublime-menu`` files are loaded
+in the same order as packages
+Menu files with the same name are concatenated,
+unless IDs are specified
+(see :ref:`below <item-ids>`).
+
+Menu files in the same package
+are loaded in lexicographical order
+starting at the root folder,
+and then traversing sub-folders in the same manner.
+
+As a special case,
+menu items from the *User* package
+declared in the standard non-ID section
+are always inserted after any standard items
+from other packages.
+
+
+.. _item-ids:
+
+Item IDs
+********
+
+When a menu item specifies an ID,
+a separate section within the menu is searched for
+and, if it does not exist,
+created at the end of the menu.
+*This ID lookup is forward-going only.*
+
+The ID determines the section's name
+and the menu item with the matching ID
+will be the first item in this section.
+All following items in the file
+will then be appended to the ID's section,
+until another item with an ID is found.
+
+If two menu items
+from different ``.sublime-menu`` files
+reference the same item via ID,
+Sublime Text will override the item's previous parameters
+with the new parameters,
+if there are any.
+Child elements in a sub menu are appended
+and do not override.
+All following items are then appended to the ID's section,
+until another item with an ID is found.
+
+It is common practice
+to assign IDs to separators
+and items having a submenu,
+so that other packages or the user themselves
+can easily customize the menu.
+This allows appending items to sections introduced by separators
+and appending items to submenus.
+
+.. note::
+
+   Due to the strict forward lookup,
+   it is possible to have
+   *multiple different items with the same ID*
+   in one menu.
+
+   Because of the potential confusion this may cause,
+   it is discouraged.
+
+   Example:
+   The following three IDs
+   are defined in a menu, in this order:
+   ``test, test2, test``.
+
+   The item with the second "test" ID
+   can then be targeted
+   using the following ID combinations:
+   ``test, test2, test``; ``test, test`` or ``test2, test``.
+
+
+Submenus
+========
+
+Every menu item can have a submenu.
+Hovering the mouse pointer
+over a menu item with a submenu
+will reveal the items grouped under it.
+submenus are independent menus
+with their own ID hierarchy.
+
+In order to extend a submenu
+from a different menu file,
+an ID must be specified in both places
+to target the correct item.
+
+
+The Main Menu
+=============
+
+Unlike all other menus,
+the Main Menu's root menu
+represents the menu items in the menu bar
+(File, Help, etc.).
+
+
+Interface For Commands
+======================
+
+Each menu item can be dynamically
+
+#. hidden,
+#. disabled,
+#. checked, or
+#. assigned a different caption.
+
+For this,
+commands must implement the required methods in their class.
+The methods are either called with the same arguments
+as the actual command would be

```

Comment:
Yes. But if there is an error calling the method with the arguments, it would be called without any. This is important information but rather hard to convey.